File: services/src/main/java/ru/katrine/bookshop/model/AuthorModel.java
package ru.katrine.bookshop.model;

import ru.katrine.bookshop.core.BaseModel;

import javax.persistence.*;
import javax.xml.bind.annotation.XmlRootElement;

@XmlRootElement
@Entity
@Table(name = "authors")
@SequenceGenerator(name = "author_id_seq", sequenceName = "author_id_seq", allocationSize = 1)
public class AuthorModel extends BaseModel {

    @Id
    @Column(name = "author_id")
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "author_id_seq")
    private Integer id;

    @Column(name = "first_name")
    private String firstName;

    @Column(name = "last_name")
    private String lastName;

    @Column(name = "middle_name")
    private String middleName;

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public String getMiddleName() {
        return middleName;
    }

    public void setMiddleName(String middleName) {
        this.middleName = middleName;
    }
}

File: web/src/app/book-editor/edit-book/_models/author.model.ts
export class AuthorModel {
  id: number;
  firstName: string;
  lastName: string;
  middleName: string;

  constructor(data){
    this.id = data.id;
    this.firstName = data.firstName;
    this.lastName = data.lastName;
    this.middleName = data.middleName;
  }

  getName(): string {
    return this.lastName + ' ' + this.firstName + ' ' + this.middleName;
  }

  getNameWithSeparators(): string{
    return this.lastName + '-' + this.firstName + '-' + this.middleName;
  }

  getShortName(){
    return this.lastName + ' ' + this.firstName[0] + '. ' + this.middleName[0] + '.';
  }
}
